netket/models/autoreg.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+from typing import Any, Callable, Iterable, Tuple, Union
+
+from flax import linen as nn
+from jax import numpy as jnp
+from netket.nn import MaskedDense1D
+from netket.nn.initializers import lecun_normal, zeros
+from netket.utils.types import Array, DType, NNInitFunc, PyTree
+
+
+class ARNN(nn.Module):
+    """Base class for autoregressive neural networks."""
+    def init_sample(self, size: Iterable[int],
+                    dtype: DType) -> Tuple[Array, PyTree]:
```

Comment:
Why do you need this? I do not understand.
This should be handled by [flax init logic](https://flax.readthedocs.io/en/latest/flax.linen.html#flax.linen.init). 
Does this suffice or there's a problem?